### ISSUE TITLE
Crash when delete all instruments and Undo (one score)

### ISSUE THREAD

**Issue by DmitryArefiev** [musescore/MuseScore#18733]
### Issue type

Crash or freeze

### Steps to reproduce

1. Open [Smoke Test 11.zip](https://github.com/musescore/MuseScore/files/12111574/Smoke.Test.11.zip)
2. In Instruments panel, Select all instruments and delete
3. Undo

### Screenshots/Screen recordings

https://github.com/musescore/MuseScore/assets/90187801/a57deaea-0fec-43cd-bb77-162a48668658

### MuseScore Version

4.1.1 nightly builds

### Regression

Yes, this used to work in a previous version of MuseScore 4.x

### Operating system

Windows10

**Comment by DmitryArefiev** [comment]
@zacjansheski Can you try that on score on your side please? 
It **crashes in 4.1.1** build and **doesn't crash in 4.1 release build**

**Comment by zacjansheski** [comment]
Confirmed in Nightly, no crash in 4.1

[InstrumentsDeleteUndoCrash.zip](https://github.com/musescore/MuseScore/files/12111855/InstrumentsDeleteUndoCrash.zip)

**Comment by cbjeukendrup** [comment]
It looks like this is somehow caused by #18279.

**Comment by RomanPudashkin** [comment]
And also note: the app crashes even when I try to open the attached score (with MUE_BUILD_ASAN=ON)
<img width="981" alt="Screen Shot 2023-07-20 at 20 05 25" src="https://github.com/musescore/MuseScore/assets/67867444/e716b49f-1fa7-49a3-af96-e76a0fd704c0">

**Comment by cbjeukendrup** [comment]
Yes, I noticed that too. A quick patch would be (a few lines up):
```patch
-        if (ac && ac->subtype() == prevSubtype && note->line() == prevLine) {
+        if (ac && (staff->isTabStaff(tick) || (ac->subtype() == prevSubtype && note->line() == prevLine))) {
```
but I've no idea if that is good or that something more rigorous is needed.

**Comment by DmitryArefiev** [comment]
Fixed in #18742
